src/main/java/de/undercouch/gradle/tasks/download/DownloadAction.java

```diff
@@ -168,7 +171,23 @@ private void execute(URL src, HttpClientFactory clientFactory) throws IOExceptio
                         + "progress will not be displayed.");
             }
         }
-        
+
+        if ("file".equals(src.getProtocol())) {
+            if (destFile.exists() && !overwrite) {
```

Comment:
This should never happen because it's the same condition as in line 138. Please check.